### ISSUE TITLE
Can't set a reference to EntityMDMD or AttributeMDMD

### ISSUE THREAD

**Issue by tommydeboer** [molgenis/molgenis#4424]
### To reproduce
- Make a new ```MetaData``` class with a reference to ```EntityMetaDataMetaData``` or ```AttributeMetaDataMetaData```
```java
public class NewMetaData extends DefaultEntityMetaData
{

        ...

	public NewMetaData()
	{
		super(ENTITY_NAME);

		addAttribute(IDENTIFIER).setIdAttribute(true).setNillable(false).setDataType(STRING);
		addAttribute(REFERENCE).setNillable(false).setDataType(MolgenisFieldTypes.XREF)
				.setRefEntity(EntityMetaDataMetaData.INSTANCE);
	}
}
```
- Start the app

### Expected
The app starts up normally and the references work.

### Actual
```java
org.molgenis.data.MolgenisDataException: Could not resolve dependencies of entities [NewMetaData] are there circular dependencies?
	at org.molgenis.util.DependencyResolver.resolve(DependencyResolver.java:103)
	at org.molgenis.data.meta.MetaDataServiceImpl.onApplicationEvent(MetaDataServiceImpl.java:464)
	at org.molgenis.data.meta.MetaDataServiceImpl.onApplicationEvent(MetaDataServiceImpl.java:1)
```

**Comment by tommydeboer** [comment]
Fixed ```DependencyResolver.resolve()``` by removing dependencies on ```entities``` and ```attributes``` (we know these entities will be available anyhow). However, at second startup, the app crashes. The ```entityMetaDataCache``` in ```EntityMetaDataRepository``` does not include ```entities``` and ```attributes``` so any references to those entities aren't added. This is tricky to fix, so I'll put this one on hold.

**Comment by dennishendriksen** [comment]
Fixed in 2.0.0-SNAPSHOT, see for example FileIngestMetaData:
```
addAttribute(ENTITY_META_DATA).setDataType(XREF).setRefEntity(entityTypeMetadata).setLabel("Target EntityType").setNillable(false);
```
